### ISSUE TITLE
sockets.ads compile error when using recent GNATPro versions

### ISSUE THREAD

**Issue by okellogg** [samueltardieu/adasockets#7]
GCC 7.3.1 20180313 for GNAT Pro 19.0w does not like the sockets.ads type [Buffer_Type](https://github.com/samueltardieu/adasockets/blob/ffc6c75e63e09ae82b7f023f40cfb4fffb8510df/src/sockets.ads#L229):

```
   type Buffer_Type
     (Length : Ada.Streams.Stream_Element_Count)
   is record
      Content : Ada.Streams.Stream_Element_Array (0 .. Length);
      --  One byte will stay unused, but this does not have any consequence
      First   : Ada.Streams.Stream_Element_Offset :=
        Ada.Streams.Stream_Element_Offset'Last;
      Last    : Ada.Streams.Stream_Element_Offset := 0;
   end record;
``` 

Really just a warning but since `-gnatg` is used, it's turned into an error:

```
if [ -f `echo sockets.ads | sed -e 's/s$/b/'` ]; then \
  /bin/sh ../libtool --tag=CC --mode=compile \
  ../support/adacompiler gnatmake -cargs -O2 -gnatpng  -I. -I. -largs  -margs \
  `echo sockets.ads | sed -e 's/s$/b/'`; \
else \
  /bin/sh ../libtool --tag=CC --mode=compile \
  ../support/adacompiler gnatmake -cargs -O2 -gnatpng  -I. -I. -largs  -margs sockets.ads; \
fi
libtool: compile:  ../support/adacompiler gnatmake -cargs -O2 -gnatpng -I. -I. -largs -margs sockets.adb  -fPIC -DPIC -o .libs/sockets.o
gcc -c -O2 -gnatpng -I. -I. -fPIC -DPIC sockets.adb
sockets.ads:232:07: warning: record layout may cause performance issues
sockets.ads:232:07: warning: component "Content" whose length depends on a discriminant
sockets.ads:232:07: warning: comes too early and was moved down
gnatmake: "sockets.adb" compilation error
mv sockets.o .libs/sockets.o
mv: cannot stat ‘sockets.o’: No such file or directory
Makefile:754: recipe for target 'sockets.lo' failed
make[3]: *** [sockets.lo] Error 1
```

**Comment by samueltardieu** [comment]
Indeed, this is a good observation. I'll reorder the fields, thanks Oliver!

**Comment by samueltardieu** [comment]
Thanks, I've released AdaSockets 1.11.1 with the reordering.

**Comment by okellogg** [comment]
Wow, this was ultra fast :+1: 
Many thanks.
